### ISSUE TITLE
writeFilesProcessor never exits

### ISSUE THREAD

**Issue by tommck** [angular/dgeni#161]
I was experiencing this with my own app, so I downloaded the latest copy of this too: https://github.com/pvadam/dgeni-example

When running "gulp", it just never exits the process after getting to writeFilesProcessor.

**Comment by petebacondarwin** [comment]
Hmm, I guess it is related to my rebuild in TypeScript? With what versions of dgeni does it happen?

**Comment by tommck** [comment]
This was with the latest versions.  You want me to try to roll back a few versions?

**Comment by petebacondarwin** [comment]
I just pulled down my original [`dgeni-example`](https://github.com/petebacondarwin/dgeni-example) repo updated the dgeni and dgeni-packages versions and ran it with gulp.
The `writeFilesProcessor` did not hang.
I have pushed to that repo the new package.json. Perhaps you could try with that?

**Comment by tommck** [comment]
I will try that.
Maybe it has something to do with Typescript?

**Comment by tommck** [comment]
Yes, that worked fine.

**Comment by tommck** [comment]
I just updated to the latest dgeni and dgeni packages for that branch and the problem stopped.. Guess there was a bug fixed in there somewhere.

**Comment by tommck** [comment]
So.. here's the setup.

grab the code from here: https://github.com/pvadam/dgeni-example/tree/typescript-example

then update to the latest dgeni and dgeni-packages.

Then the writeFilesProcessor never exits

**Comment by petebacondarwin** [comment]
I will take a look tomorrow!

**Comment by tommck** [comment]
I don't want to pester (but I will).

Did you get a chance to look at this?

**Comment by petebacondarwin** [comment]
Fair enough - today is not what was going to be tomorrow. I'll look in the next hour...

**Comment by tommck** [comment]
Thanks so much. I really appreciate it

**Comment by petebacondarwin** [comment]
I just did the following (node version 6.9.5, OS/X Sierra)

```bash
git clone https://github.com/pvadam/dgeni-example
cd dgeni-example/
git checkout origin/typescript-example
git checkout -b typescript-example
npm install
npm install dgeni@latest dgeni-packages@latest
gulp
```

And I get a massive long output (as expected since it is set to log "silly")
But it does not hang on the `writeFilesProcessor`....

**Comment by petebacondarwin** [comment]
The last few lines of the log look like:

```
info:    running processor: writeFilesProcessor
silly:   writing file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/src/app/index.html
silly:   writing file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/src/app/helloDgeni/index.html
debug:   Document "src/app/MyApp.greet, member" has no outputPath.
silly:   writing file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/index.html
silly:   writing file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/Logger/index.html
debug:   Document "/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/Logger.log, member" has no outputPath.
debug:   written file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/src/app/index.html
debug:   written file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/src/app/helloDgeni/index.html
debug:   written file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/index.html
debug:   written file /Users/pete/dev/dgeni/temp/dgeni-example/build/partials/modules/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/Logger/index.html
info:    running processor: checkAnchorLinksProcessor
silly:   checking file hrefs=[], names=[], path=/src/app, outputPath=partials/modules/src/app/index.html
silly:   checking file hrefs=[], names=[], path=/src/app/helloDgeni, outputPath=partials/modules//src/app/helloDgeni/index.html
silly:   checking file hrefs=[], names=[], path=/Users/pete/dev/dgeni/temp/dgeni-example/src/logger, outputPath=partials/modules//Users/pete/dev/dgeni/temp/dgeni-example/src/logger/index.html
silly:   checking file hrefs=[], names=[], path=/Users/pete/dev/dgeni/temp/dgeni-example/src/logger/Logger, outputPath=partials/modules///Users/pete/dev/dgeni/temp/dgeni-example/src/logger/Logger/index.html
```

**Comment by tommck** [comment]
I'm on windows and I'm running Node 6, but an older version.. lemme update and try again (because your commands *do* lockup on writeFilesProcessor for me)

**Comment by tommck** [comment]
Upgraded to node 6.10.0 and it still locks up on writeFilesProcessor :(

Anyone else who can test this on windows?

**Comment by tommck** [comment]
I'm on Windows 10 in case that matters

**Comment by tommck** [comment]
Last few lines

```
info:    running processor: unescapeCommentsProcessor
info:    running processor: inlineTagProcessor
info:    running processor: writeFilesProcessor
silly:   writing file C:/dev/delete/dgeni-example/build/partials/modules/src/app/index.html
silly:   writing file C:/dev/delete/dgeni-example/build/partials/modules/src/app/helloDgeni/index.html
debug:   Document "src/app/MyApp.greet, member" has no outputPath.
silly:   writing file C:/dev/delete/dgeni-example/build/partials/modules/C:/dev/delete/dgeni-example/src/logger/index.html
silly:   writing file C:/dev/delete/dgeni-example/build/partials/modules/C:/dev/delete/dgeni-example/src/logger/Logger/index.html
debug:   Document "C:/dev/delete/dgeni-example/src/logger/Logger.log, member" has no outputPath.
debug:   written file C:/dev/delete/dgeni-example/build/partials/modules/src/app/index.html
debug:   written file C:/dev/delete/dgeni-example/build/partials/modules/src/app/helloDgeni/index.html
```

**Comment by gkalpak** [comment]
I tried this on Windows 10 (node v6.9.5) and it does hang with the same output as @tommck (except the last two lines are swapped 😕 ).

**Comment by petebacondarwin** [comment]
OK. I better dig my windows machine out from a cupboard...

**Comment by tommck** [comment]
I'm sooooo glad! Thought I was crazy there for a while!

Thank you to so much for the help

**Comment by petebacondarwin** [comment]
Fixed in 0.17.1
